### ISSUE TITLE
Which global should we use for throwing exceptions?

### ISSUE THREAD

**Issue by asajeffrey** [servo/servo#15328]
In the code generator for exceptions, we generate, `throw_dom_exception(cx, &this.global(), e);` which uses the global for the current `this` object. Should this be the current global?

Spec issue: https://github.com/heycam/webidl/issues/292

IRC chat with @jdm, @nox and @bzbarsky: logs.glob.uno/?c=mozilla%23servo&s=1+Feb+2017&e=1+Feb+2017#c603463

**Comment by bzbarsky** [comment]
You very much want the current global, I expect.  Especially for the cross-origin case, where using the global of `this` would create an exception the caller can't even interact with.

**Comment by asajeffrey** [comment]
Going back over the IRC conversation, the conclusion is: http://logs.glob.uno/?c=mozilla%23servo&s=1+Feb+2017&e=1+Feb+2017#c603803

TL;DR: we need XOWs to make this work, which is issue #2382. cc @avadacatavra

**Comment by asajeffrey** [comment]
Closed in favour of #2382.
